js-practive/js/main.js

```diff
@@ -1 +1,260 @@
-// Place your code here, or you can split it how you what. It's up to you 😎
\ No newline at end of file
+// console.log(event)
+    // 1.очищення всієї сторінки 
+    // 2.умови, що знаходяться в хещі та їх валідація (асинхронний запит який вибудовує сторінку) (вбудувати сторінку )
+let usersTable
+let createButton
+let userForm 
+
+setUsersTable()
+setHtmlElementUserForm()
+
+function setUsersTable() {
+    usersTable = document.getElementById('users-table-list');
+    createButton = document.getElementById('create');
+
+    createButton.addEventListener('click', goToUserForm);
+    usersTable.addEventListener('click', onUsersTableClick);
+}
+
+function setHtmlElementUserForm() {
+    userForm = document.getElementById('users-edit');
+    userForm.addEventListener('click', onUserFormClick);
+}
+
+readingHashState();
+
+function readingHashState() {
+    addEventListener('popstate', (event) => {
+            
+        switch (location.hash.split("=")[0]) {
+            case '#user-form': {
+                if (window.location.hash.split("=")[1]) {
+                    fetch('/user-form.html')
+                    .then((res) => res.text())
+                    .then((stringHtml) => document.body.innerHTML = stringHtml)
+                    .then(() => getSingleUser(window.location.hash.split("=")[1]))
+                    .then((user) => {
+                        setHtmlElementUserForm()
+                        setUserData(user)
+                    })
+    
+                } else {
+                    fetch('/user-form.html')
+                    .then((res) => res.text())
+                    .then((stringHtml) => document.body.innerHTML = stringHtml)
+                    .then(() => setHtmlElementUserForm())
+                    .then(() => {
+                        if(getDataOnLocalStorage()) {
+                            setUserData(getDataOnLocalStorage())
+                        }
+                    })
+                }            
+                break
+            }
+            case '': {
```

Comment:
maybe better in this case to use `default`

Reply:
but we can have another case, thats why i didn`t used default